components/scripts/gradle/03-validate-local-build-caching-different-locations.sh

```diff
@@ -157,9 +171,45 @@ execute_second_build() {
      clean ${tasks}
 }
 
+print_gradle_command() {
+  if [[ "${build_scan_publishing_mode}" == "on" ]]; then
+    info "./gradlew --build-cache --scan -Dscan.tag.${EXP_SCAN_TAG} -Dscan.value.runId=${RUN_ID} clean ${tasks}$(print_extra_args)"
+  else
+    info "./gradlew --build-cache -Dscan.dump -Dscan.tag.${EXP_SCAN_TAG} -Dscan.value.runId=${RUN_ID} clean ${tasks}$(print_extra_args)"
+  fi
+}
+
 fetch_build_cache_metrics() {
-  read_build_scan_metadata
-  fetch_and_read_build_scan_data build_cache_metrics_only "${build_scan_urls[@]}"
+  if [ "$build_scan_publishing_mode" == "on" ]; then
+    read_build_scan_metadata
+    fetch_and_read_build_scan_data build_cache_metrics_only "${build_scan_urls[@]}"
+  else
+    find_build_scan_dumps
+    read_build_scan_dumps
+  fi
+}
+
+find_build_scan_dumps() {
+  find_build_scan_dump "first"
+  find_build_scan_dump "second"
+}
+
+find_build_scan_dump() {
+  local build_name="$1"
+  local build_dir="${EXP_DIR}/$build_name-build_${project_name}"
```

Comment:
Does this start deep enough in the folder hierarchy to not find a scan dump from a previous experiment run?

Reply:
I think so, but just wanted to ask for confirmation.

Reply:
Yes, it does. This is the exact path to the root project directory (e.g. `.data/02-validate-local-build-caching-same-location/20221109T135440-636c0580/first-build_java-ordered-properties`).

However, now I'm wondering how this should work when `-p` / `--project-dir` is used? 

cc @jthurne

Reply:
-p should only influence the dir from which the project is invoked. I think all is fine. But, @jthurne please weigh in here.

Reply:
If the user provides `-p/--project-dir, before invoking the build tool, the scripts will cd to the directory given by `-p` within the repo, and then invoke Gradle or Maven.  This means that any output generated by Gradle will also be in the subdirectory, including the build scan dump.   As such, I do think there may be a problem here.

There are two ways we could correct this:

  1. If `project_dir` is set, include it in the path where we look for the build scan dump.
  2. remove the `-maxdepth` argument from the find command so that it finds build scan dumps that may be in a subdirectory.
 
The first option is most likely the "safest".

Reply:
Thanks for the input. I will address this in a different PR. I have created #217 for this.